activesupport/lib/active_support/core_ext/hash/deep_compact.rb

```diff
@@ -0,0 +1,46 @@
+class Hash
+  def deep_compact(remove_blank: false)
```

Comment:
Rails has compact_blank. IMO this should be deep_compact_blank (or to add two methods, instead of a boolean arg)